cluster.go

```diff
@@ -358,6 +381,53 @@ func parseShards(shards RedisMessage, defaultAddr string, tls bool) map[string]g
 	return groups
 }
 
+func (c *clusterClient) runClusterTopologyRefreshment() {
+	ticker := time.NewTicker(c.opt.ClusterTopologyRefreshmentOption.ScanInterval)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-c.refreshStop:
+			return
+		case <-ticker.C:
+			result, err := c.getClusterTopology()
+			if err != nil {
+				c.lazyRefresh()
+				continue
+			}
+
+			groups := result.parse(c.opt.TLSConfig != nil)
+
+			conns := c.newConns(groups)
+
+			isChanged := false
+			c.mu.RLock()
+			// check if the new topology is different from the current one
+			for addr, cc := range conns {
+				old, ok := c.conns[addr]
+				if !ok || old.replica != cc.replica {
+					isChanged = true
+					break
+				}
+			}
+
+			// check if the current topology is different from the new one
+			if !isChanged {
+				for addr := range c.conns {
+					if _, ok := conns[addr]; !ok {
+						isChanged = true
+						break
+					}
+				}
+			}
+			c.mu.RUnlock()
```

Comment:
Why not put this check in the lazyRefresh? Then we can simply call lazyRefresh periodically here.

Reply:
I don't understand. this check logic is for refresh only topology is changed. if you want to make a new method, i added it. [027671b](https://github.com/redis/rueidis/pull/638/commits/027671b8a08e824c58cc35a069c8cf980c1abc49)

Reply:
Ok, how about this?

[b075e69](https://github.com/redis/rueidis/pull/638/commits/b075e699c15affca0f3595532edcf9120cf7b0ce)